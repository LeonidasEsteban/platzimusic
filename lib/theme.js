const theme = {
  font: {
    title: 'Quicksand, Helvetica, Arial',
  },
  flexboxgrid: {
    gutterWidth: 0,
    container: {
      md: 64,
      lg: 64,
    },
  },
  color: {
    grayA: '#e1e4ef',
    grayB: '#a6acc1',
    grayC: '#4b4e5a',
    blue: '#5179ff',
    pink: '#ea83ee',
  },
};

export default theme;

