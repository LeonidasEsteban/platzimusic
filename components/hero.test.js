/* global it, expect, describe */
import React from 'react';
import { render } from 'react-dom';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import Hero from './hero';
import theme from '../lib/theme';

describe('Test Hero component', () => {
  it('should render without small', () => {
    const div = document.createElement('div');
    render(
      <ThemeProvider theme={theme}>
        <Hero>
          <div>testing without small</div>
        </Hero>
      </ThemeProvider>,
      div,
    );
  });

  it('should render with small', () => {
    const div = document.createElement('div');
    render(
      <ThemeProvider theme={theme}>
        <Hero small>
          <div>testing with small</div>
        </Hero>
      </ThemeProvider>,
      div,
    );
  });

  it('should render without small the expected HTML', () => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <Hero>
          <div>testing without small</div>
        </Hero>
      </ThemeProvider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with small the expected HTML', () => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <Hero small>
          <div>testing with small</div>
        </Hero>
      </ThemeProvider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
