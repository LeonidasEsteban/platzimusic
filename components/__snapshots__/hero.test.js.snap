// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Hero component should render with small the expected HTML 1`] = `
<div
  className="bvTdkS"
>
  <div
    className="kfsALc"
  >
    <div>
      <h1
        className="lkuQWA"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <strong>
            Platzi
          </strong>
          Music
        </a>
      </h1>
    </div>
    <div>
      testing with small
    </div>
  </div>
</div>
`;

exports[`Test Hero component should render without small the expected HTML 1`] = `
<div
  className="eiCoQH"
>
  <div
    className="kfsALc"
  >
    <div>
      <h1
        className="leapIt"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <strong>
            Platzi
          </strong>
          Music
        </a>
      </h1>
      <h2
        className="gpBNgN"
      >
        Tu música sin límites
      </h2>
    </div>
    <div>
      testing without small
    </div>
  </div>
</div>
`;
